src/ensembl/src/content/app/entity-viewer/shared/components/entity-viewer-sidebar/entity-viewer-sidebar-toolstrip/EntityViewerSidebarToolstrip.tsx

```diff
@@ -0,0 +1,124 @@
+
+import { ReactComponent as searchIcon } from 'static/img/sidebar/search.svg';
+import { ReactComponent as bookmarkIcon } from 'static/img/sidebar/bookmark.svg';
+import { ReactComponent as shareIcon } from 'static/img/sidebar/share.svg';
+import { ReactComponent as downloadIcon } from 'static/img/sidebar/download.svg';
+
+import { RootState } from 'src/store';
+import { Status } from 'src/shared/types/status';
+import { SidebarToolstripCollection } from 'src/shared/types/sidebar-toolstrip-collection';
+import styles from 'src/shared/components/layout/StandardAppLayout.scss';
+
+export type entityViewerSidebarBarProps = {
+  isEntityViewerSidebarModalOpen: boolean;
+  isEntityViewerSidebarOpen: boolean;
+  sidebarModalView: string;
+  closeSidebarModal: () => void;
+  openSidebarModal: (sidebarModalView: string) => void;
+  toggleSidebar: (isEntityViewerSidebarOpen?: boolean) => void;
+};
+
+export const EntityViewerSidebarToolstrip = (props: entityViewerSidebarBarProps) => {
+  const toggleModalView = (selectedItem: string) => {
+    if (!props.isEntityViewerSidebarOpen) {
+      props.toggleSidebar();
+    }
+
+    if (selectedItem === props.sidebarModalView) {
+      props.closeSidebarModal();
+    } else {
+      props.openSidebarModal(selectedItem);
+    }
+  };
+
+  const getViewIconStatus = (selectedItem: string) => {
+    return selectedItem === props.sidebarModalView &&
+      props.isEntityViewerSidebarOpen
+      ? Status.SELECTED
+      : Status.UNSELECTED;
+  };
+
+  return (
+    <>
+      <ImageButton
+        status={Status.DISABLED}
+        description="Search"
+        className={styles.sidebarIcon}
+        key="search"
+        onClick={noop}
+        image={searchIcon}
+      />
+      <ImageButton
+        status={getViewIconStatus(SidebarToolstripCollection.BOOKMARKS)}
+        description="Bookmarks"
+        className={styles.sidebarIcon}
+        key={SidebarToolstripCollection.BOOKMARKS}
+        onClick={() => toggleModalView(SidebarToolstripCollection.BOOKMARKS)}
+        image={bookmarkIcon}
+      />
+      <ImageButton
+        status={Status.DISABLED}
+        description="Share"
+        className={styles.sidebarIcon}
+        key="share"
+        onClick={noop}
+        image={shareIcon}
+      />
+      <ImageButton
+        status={Status.DISABLED}
+        description="Downloads"
+        className={styles.sidebarIcon}
+        key="downloads"
+        onClick={noop}
+        image={downloadIcon}
+      />
+    </>
+  );
+};
+
+
+const mapStateToProps = (state: RootState) => ({
+  isEntityViewerSidebarModalOpen: isEntityViewerSidebarModalOpen(state),
+  isEntityViewerSidebarOpen: isEntityViewerSidebarOpen(state),
+  sidebarModalView: getEntityViewerSidebarModalView(state)
+});
+
+const mapDispatchToProps = {
+  closeSidebarModal,
+  openSidebarModal,
+  toggleSidebar
+};
+
+export default connect(mapStateToProps, mapDispatchToProps)(EntityViewerSidebarToolstrip);
```

Comment:
This component also can be converted into a non-connected component with `useSelector` and `useDispatch` like what you've done with other components above